[Bug]: Typescript error related to AccordionProps

### Package

@carbon/react

### Browser

Chrome

### Package version

v1.65.0

### React version

v18.2

### Description

I use carbon v1.65.0 (with vite and typescript) and have an accordion using AccordionProps.
When I try to run/build my client - I get these TS errors: 
`error TS2559: Type '{ children: Element[]; }' has no properties in common with type 'IntrinsicAttributes & AccordionProps'.
error TS2559: Type '{ children: Element; }' has no properties in common with type 'IntrinsicAttributes & AccordionProps'.`

Looks like passing children to Accordion isn't supported anymore.

### Reproduction/example

https://stackblitz.com/edit/github-lctgtu?file=src%2FApp.tsx

### Steps to reproduce

Create a react application with vite, typescript and carbon (latest version v1.65.0).
Create an accordion component using Carbon Accordion and use AccordionProps and the props type. Try to pass children as prop, and you will see the errors. 

<img width="665" alt="Screenshot 2024-09-10 at 20 17 59" src="https://github.com/user-attachments/assets/da24d1e6-6da2-4975-9a70-6945f6c8942e">


### Suggested Severity

Severity 3 = User can complete task, and/or has a workaround within the user experience of a given component.

### Application/PAL

_No response_

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

--- a/packages/react/src/components/Accordion/Accordion.tsx
+++ b/packages/react/src/components/Accordion/Accordion.tsx
@@ -8,7 +8,7 @@
 import cx from 'classnames';
 import { usePrefix } from '../../internal/usePrefix';
 import PropTypes from 'prop-types';
-import React, { PropsWithChildren } from 'react';
+import React, { PropsWithChildren, ReactNode } from 'react';
 import { AccordionProvider } from './AccordionProvider';
 
 export interface AccordionProps {
@@ -23,6 +23,11 @@ export interface AccordionProps {
    * the container node.
    */
   className?: string;
+
+  /**
+   * Pass in the children that will be rendered within the Accordion
+   */
+  children?: ReactNode;
 
   /**
    * Specify whether an individual AccordionItem

